switchboard/app/switchboard.hoon

```diff
@@ -0,0 +1,471 @@
+    =/  call-state  (~(get by calls.state) uuid)
+    ?~  call-state
+      `this
+    =^  cards  state
+    (remote-disconnected:helper call.u.call-state)
+    [cards this]
+    ::
+      [%call @ta ~]
+    =/  uuid  +<.path
+    =/  call-state  (~(get by calls.state) uuid)
+    ?~  call-state
+      `this
+    =^  cards  state
+    (local-disconnected call.u.call-state)
+    [cards this]
+  ==
+:: TODO: scry per-call for state
+:: TODO: scry for registered daps
+++  on-peek
+  |=  =path
+  ^-  (unit (unit cage))
+  ?+  path  (on-peek:default path)
+      :: per-call state
+      [%x %callstate @ta ~]
+    ?>  (team:title our.bowl src.bowl)
+    =/  uuid  +>-.path
+    (biff (~(get by calls.state) uuid) |=(=call-state:switchboard ``[%noun !>(call-state)]))
+      :: registered daps
+      [%x %registered ~]
+    ?>  (team:title our.bowl src.bowl)
+    ``[%noun !>(((list @tas) ~(tap in ~(key by reachable.state))))]
+  ==
+:: state advancements for watch-ack and poke-ack
+::
+:: Call ending on %kick from remote agent
+::
+:: Fact relay from remote agent (the core functionality of switchboard)
+++  on-agent
+  |=  [=wire =sign:agent:gall]
+  ^-  (quip card _this)
+  ?+  wire  (on-agent:default wire sign)
+      ::
+      [%peer-signal @ta ~]
+    =/  uuid  +<.wire
+    =/  call-state  (~(get by calls.state) uuid)
+    ?~  call-state
+      ~|  "No call with UUID {<uuid>}"  !!
+    ?+  -.sign  (on-agent:default wire sign)
+        :: check positive watch-ack, if so and %answered, ->
+        :: %connected
+        :: if negative, kick dap
+        %watch-ack
+      ?+  connection-state.u.call-state ::-::
+        ?~  +.sign
+          `this
+        =^  cards  state
+        (remote-disconnected:helper call.u.call-state)
+        [cards this]
+          :: %answered -> %connected
+          %answered
+        =^  cards  state
+        ?~  +.sign
+          (move-to-state:helper call.u.call-state %connected)
+        (remote-disconnected:helper call.u.call-state)
+        [cards this]
+      ==
+        :: SDP message, relay to dap
+        %fact
+      :_  this
+      :~
+        :*
+          %give  %fact
+          ~[/call/[uuid]]
+          cage.sign
+        ==
+      ==
+
+        :: Hang up, kick dap
+        %kick
+      =^  cards  state
+      (remote-disconnected:helper call.u.call-state)
+      [cards this]
+    ==
+      ::
+      [%ring-poke @ta ~]
+    =/  uuid  +<.wire
+    =/  call-state  (~(get by calls.state) uuid)
+    ?~  call-state
+      ~|  "No call with UUID {<uuid>}"  !!
+    ?+  -.sign  (on-agent:default wire sign)
+        :: check positive poke-ack, if so and %dialing, -> %ringing
+        :: if negative, kick dap
+        %poke-ack
+      =^  cards  state
+      ?~  +.sign
+        (move-to-state:helper call.u.call-state %ringing)
+      (remote-disconnected:helper call.u.call-state)
+      [cards this]
+    ==
+  ==
+++  on-arvo  on-arvo:default
+++  on-fail   on-fail:default
+--
+:: Helper core
+|_  bowl:gall
```

Comment:
Do you use your bowl? It doesn't look like it. If you don't, then you should just write this as a core `|%`.